workflows/prognostic_run_diags/fv3net/artifacts/query.py

```diff
@@ -0,0 +1,142 @@
+import argparse
+import asyncio
+import dataclasses
+import datetime
+import os
+import pathlib
+
+import fsspec
+
+import gcsfs
+
+
+@dataclasses.dataclass
+class Step:
+    fs: fsspec.AbstractFileSystem
+    path: pathlib.Path
+    date_created: datetime.date
+    project: str
+    tag: str
+    step: str
+
+
+async def flat_gather(promises):
+    out = await asyncio.gather(*promises)
+    return sum(out, [])
+
+
+async def _get_runs_with_tag(f, tag, date, project):
+    out = []
+    for step in await _list(f, tag):
+        path = pathlib.Path(step)
+        if path.name in ["fv3gfs_run", "fv3gfs_run_diagnostics"]:
```

Comment:
Seems like it'd be useful to be able to find `trained_models` and `offline_diags` as well.